flowmachine/flowmachine/features/subscriber/home_location_monthly.py

```diff
@@ -0,0 +1,198 @@
+from typing import List, Union
+from flowmachine.core.spatial_unit import AnySpatialUnit
+from datetime import datetime, timedelta
+from flowmachine.core.query import Query
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import rrule, DAILY
+from copy import copy
+
+from flowmachine.features.subscriber.daily_location import daily_location
+from flowmachine.features.subscriber.modal_location import ModalLocation
+from flowmachine.features.subscriber.unique_active_subscribers import (
+    UniqueActiveSubscribers,
+    ActiveSubscribers,
+)
+from flowmachine.features.subscriber.last_location import LastLocation
+from flowmachine.features.subscriber.location_visits import LocationVisits
+
+
+class HomeLocationMonthly(Query):
+
+    """For each subscriber, if their modal last location was their last location on more than X days this month
+    (choose X=15 here), then choose this as their home location. If not, but it was their last location on more than Y
+     days this month (Y<X; choose Y=10 here) AND it is the same as their “reference location” (in our case reference
+     location is modal location and last location on more than X days last month), then choose this as their home
+     location. Otherwise set their home location to “unlocatable” (a special value, not NULL)."""
+
+    def __init__(
+        self,
+        window_start: Union[str, datetime],
+        window_stop: Union[str, datetime],
+        spatial_unit: AnySpatialUnit,
+        home_this_month: int,
+        home_last_month: int,
```

Comment:
The names 'home_this_month' and 'home_last_month' don't really capture what these parameters are - perhaps something like 'threshold_for_non_reference_location' and 'threshold_for_reference_location' would be more descriptive?

Reply from @Thingus:
> I've been calling them `known_location_threshold` and `unknown_location_threshold`, or just `X` and `Y` after the docstring - but none of these quite capture what it is that those are doing. Will think on it.